Packages/ChatSecure/ChatSecure/Classes/WebRTCJanus/Janus.swift

```diff
@@ -174,74 +183,79 @@ class Janus: NSObject {
 	}
 	
 	func send(message msg: [String: Any], jsep: [String: Any]?, handleId: NSNumber, callback: @escaping PluginRequestCallback) {
-		if sessionId == 0 || janusWebSocket == nil {
-			let data = ["error_code": NSNumber(value: -1), "error": "sessionID januswebsocket = 0"] as [String: Any]
-			callback(data, nil)
-			return
-		}
-		
-		var transaction = randomString(withLength: 12)
-		while janusTransactions.keys.contains(transaction) {
-			transaction = randomString(withLength: 12)
-		}
-		var params: [String: Any]
-		if let jsep = jsep {
-			params = ["janus": JanusMessage.message.rawValue,
-					  "body": msg,
-					  "jsep": jsep,
-					  "transaction": transaction,
-					  "session_id": NSNumber(value: sessionId),
-					  "handle_id": handleId]
-		} else {
-			params = ["janus": JanusMessage.message.rawValue,
-					  "body": msg,
-					  "transaction": transaction,
-					  "session_id": NSNumber(value: sessionId),
-					  "handle_id": handleId]
-		}
-		
-		if let token = self.token {
-			params["token"] = token
-		}
-		if let secret = self.apiSecret {
-			params["apisecret"] = secret
-		}
-		let callbackJanus: JanusRequestCallback = { msg in
-			if let janus = msg["janus"] as? String, janus == "error",
-			   let error = msg["error"] as? [String: Any] {
-				let data: [String: Any] = ["error_code": error["code"] ?? "", "error_reason": error["reason"] ?? ""]
+		mainque.async {
+			if self.sessionId == 0 || self.janusWebSocket == nil {
+				let data = ["error_code": NSNumber(value: -1), "error": "sessionID januswebsocket = 0"] as [String: Any]
 				callback(data, nil)
-			} else if let pluginData = msg["plugindata"] as? [String: Any],
-					  let data = pluginData["data"] as? [String: Any] {
-				let jsep = msg["jsep"] as? [String: Any]
-				callback(data, jsep)
+				return
+			}
+			
+			var transaction = randomString(withLength: 12)
+			while self.janusTransactions.keys.contains(transaction) {
+				transaction = randomString(withLength: 12)
+			}
+			var params: [String: Any]
+			if let jsep = jsep {
+				params = ["janus": JanusMessage.message.rawValue,
+						  "body": msg,
+						  "jsep": jsep,
+						  "transaction": transaction,
+						  "session_id": NSNumber(value: self.sessionId),
+						  "handle_id": handleId]
 			} else {
-				debugPrint("send no handle callback")
+				params = ["janus": JanusMessage.message.rawValue,
+						  "body": msg,
+						  "transaction": transaction,
+						  "session_id": NSNumber(value: self.sessionId),
+						  "handle_id": handleId]
+			}
+			
+			if let token = self.token {
+				params["token"] = token
+			}
+			if let secret = self.apiSecret {
+				params["apisecret"] = secret
 			}
+			let callbackJanus: JanusRequestCallback = {[weak self] msg in
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/e5e0a713c2a0f396e41d6374bb8f07759fda2936/Packages/ChatSecure/ChatSecure/Classes/WebRTCJanus/Janus.swift#L219"></span>Unused reference self in a capture list should be removed.<br />
<code>unused_capture_list</code> <code>Janus.swift:219</code></td>
    </tr>
  </tbody>
</table>